### PR TITLE
Adds latest/earliest block fetch to eth_getBlockByNumber

### PR DESCRIPTION
**Pull request by otherview**
### Why is this change needed?

- https://github.com/obscuronet/obscuro-internal/issues/596
- Fixes the expected strings returns in the block number field

### What changes were made as part of this PR:

- functional

### CODE REVIEW

File: go/host/rpc/clientapi/client_api_eth.go [hunk 1]
```diff
@@ -49,11 +49,24 @@ func (api *EthereumAPI) GetBalance(_ context.Context, encryptedParams common.Enc
 
 // GetBlockByNumber returns the rollup with the given height as a block. No transactions are included.
 func (api *EthereumAPI) GetBlockByNumber(_ context.Context, number rpc.BlockNumber, _ bool) (map[string]interface{}, error) {
+	var rollupHeaderWithHashes *common.HeaderWithTxHashes
```
Comment (joeldudley) [inline go/host/rpc/clientapi/client_api_eth.go:52]: Nitpick - don't think you need to predeclare this.

File: go/host/rpc/clientapi/client_api_eth.go [hunk 2]
```diff
@@ -49,11 +49,24 @@ func (api *EthereumAPI) GetBalance(_ context.Context, encryptedParams common.Enc
...
+
+	switch number {
+	case rpc.LatestBlockNumber:
+		return headerWithHashesToBlock(api.host.DB().GetCurrentRollupHead()), nil
+	case rpc.EarliestBlockNumber:
+		rollupHeaderWithHashes = api.host.DB().GetRollupHeader(gethcommon.BigToHash(big.NewInt(0)))
```
Comment (joeldudley) [inline go/host/rpc/clientapi/client_api_eth.go:58]: Just double-checking - we def want the genesis rollup, not the first produced rollup, right?

**Review by joeldudley — APPROVED** [review]
LGTM (but left some comments).

File: go/host/rpc/clientapi/client_api_eth.go [hunk 3]
```diff
@@ -49,11 +49,24 @@ func (api *EthereumAPI) GetBalance(_ context.Context, encryptedParams common.Enc
 
 // GetBlockByNumber returns the rollup with the given height as a block. No transactions are included.
 func (api *EthereumAPI) GetBlockByNumber(_ context.Context, number rpc.BlockNumber, _ bool) (map[string]interface{}, error) {
+	var rollupHeaderWithHashes *common.HeaderWithTxHashes
+
+	switch number {
```
Comment (tudor-malene) [inline go/host/rpc/clientapi/client_api_eth.go:54]: just out of curiosity.
Is this something that geth does as well?

**Review by tudor-malene — COMMENTED** [review]
only a q

File: go/host/rpc/clientapi/client_api_eth.go (hunk 1 above)
Comment (otherview) [reply go/host/rpc/clientapi/client_api_eth.go:52]: Good spot, fixed.

File: go/host/rpc/clientapi/client_api_eth.go (hunk 3 above)
Comment (otherview) [reply go/host/rpc/clientapi/client_api_eth.go:54]: Yep!
https://github.com/ethereum/go-ethereum/blob/c6dcd018d25998201eff1e1e264c9080847d92df/rpc/types.go#L77-L110

**Pull request merged**